auction-server/src/auction/service/mod.rs

```diff
@@ -272,6 +272,132 @@ pub enum ServiceEnum {
     Svm(Service<Svm>),
 }
 
+#[cfg(test)]
+pub use {
+    mock_service::MockService as StatefulMockAuctionService,
+    service_wrapper::MockService,
+};
+
+#[cfg(test)]
+mod service_wrapper {
+    use super::*;
+    pub use mock_service::MockService as InnerMockService;
+    #[derive(Clone)]
+    pub struct MockService<T: ChainTrait>(pub Arc<InnerMockService<T>>);
+
+    impl<T: ChainTrait> MockService<T> {
+        pub fn new(mock: InnerMockService<T>) -> Self {
+            Self(Arc::new(mock))
+        }
+    }
+
+    impl<T: ChainTrait> std::ops::Deref for MockService<T> {
+        type Target = InnerMockService<T>;
+        fn deref(&self) -> &Self::Target {
+            &self.0
+        }
+    }
+}
+
+#[cfg(test)]
+mod mock_service {
+    use {
+        super::*,
+        crate::{
+            api::RestError,
+            kernel::{
+                contracts::{
+                    MulticallData,
+                    MulticallStatus,
+                },
+                entities::PermissionKey,
+            },
+        },
+        ethers::contract::FunctionCall,
+        mockall::mock,
+        solana_sdk::{
+            instruction::CompiledInstruction,
+            transaction::VersionedTransaction,
+        },
+    };
+
+    mock! {
+        pub Service<T: ChainTrait> {
```

Comment:
```suggestion
        pub ServiceInner<T: ChainTrait> {
```
for more consistency?